Automattic/wp-calypso
# Atomic Hosting: Restore DB password

**DO NOT MERGE**
Requires D34795-code

#### Changes proposed in this Pull Request

Adds a link to the Atomic Hosting Dashboard for restoring the database password.

<img width="961" alt="Screen Shot 2019-10-31 at 15 30 36" src="https://user-images.githubusercontent.com/1233880/67955932-97910880-fbf3-11e9-8583-49de83bcc02c.png">
<img width="756" alt="Screen Shot 2019-10-31 at 15 30 46" src="https://user-images.githubusercontent.com/1233880/67955939-99f36280-fbf3-11e9-98b5-621fbdf85e63.png">

#### Testing instructions

* Apply D34795-code and sandbox the API.
* In Calypso, switch to an Atomic Site.
* Go to "Manage > SFTP & MySQL".
* Break your site by opening phpMyAdmin and running `SET PASSWORD = PASSWORD( 'password' )`;
* Visit your site and confirm there is a DB error preventing the access.
* Back in Calypso, go to "Manage > SFTP & MySQL" again.
* Confirm there is a new message under the "Open phpMyAdmin" button that contains a link for restoring the database password.
* Click on that link.
* Verify a dialog is displayed asking for confirmation.
* Click on "Restore".
* Wait for the success notice.
* Visit your site and confirm you can access the site now.

Fixes #36283
